mern-ecommerce/controllers/auth.js

```diff
@@ -80,3 +80,13 @@ exports.isAdmin = (req, res, next) => {
   }
   next();
 };
+
+exports.checkUser = (req, res, next) => {
+  let user = req.profile && req.auth && req.profile._id == req.auth._id;
+  if (!user) {
+    return res.status(403).json({
+      error: 'Access denied',
```

Comment:
Error message is not descriptive. Consider adding more context or details about the error.